idl/chromadb/proto/query_executor.proto

```diff
@@ -0,0 +1,92 @@
+syntax = "proto3";
+
+package chroma;
+
+import "chromadb/proto/chroma.proto";
+
+message ScanOperator {
+    Collection collection = 1;
+    string knn_id = 2;
+    string metadata_id = 3;
+    string record_id = 4;
+}
+
+message FilterOperator {
+    optional UserIds ids = 1;
+    optional Where where = 2;
+    optional WhereDocument where_document = 3;
+}
+
+message KNNOperator {
+    repeated Vector embeddings = 1;
+    uint32 fetch = 2;
```

Comment:
preference for using the names our api uses.

Reply:
Will update the language in a separate PR after this stack is merged